v2/contribute/decisions/oauth/0009-account-linking.mdx

```diff
@@ -0,0 +1,40 @@
+---
+id: "0009"
+title: Use account linking if using ST as a third-party provider
+hide_title: true
+---
+
+import DecisionHeader from "/src/components/decisionLogs/DecisionHeader"
+import ArgumentPro from "/src/components/decisionLogs/ArgumentPro"
+import ArgumentCon from "/src/components/decisionLogs/ArgumentCon"
+import ArgumentNeut from "/src/components/decisionLogs/ArgumentNeut"
+
+# Use account linking if using ST as a third-party provider
+
+<DecisionHeader status="proposed" lastUpdate="2023-05-11" created="2023-05-11" deciders={["rishabhpoddar", "porcellus"]} proposedBy={["porcellus"]} />
+
+## Context and Problem Statement
+
+We will provide support for using ST as a provider in the ThirdParty (and combination) recipes. Ideally, we'd like to re-use the user id we have in the original.
+
+## Considered Options
+
+* Use user id mapping
+* Use account linking
+
+## Decision Outcome
+
+Chosen option: Use account linking if using ST as a third-party provider
+
+* Doesn't require using separate apps (as in multitenancy) for the provider and the consumer.
+
```

Comment:
One important point here that we had discussed is that this should continue to work even if the account linking feature flag is switched off - this exception case can be easily done on the core side by not checking for the license key features when doing account linking for this use case.